spark-bigquery-dsv2/spark-3.4-bigquery-lib/src/test/java/com/google/cloud/spark/bigquery/v2/TimestampNTZTypeConverterTest.java

```diff
@@ -0,0 +1,109 @@
+
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.storage.v1.CivilTimeEncoder;
+import com.google.protobuf.DescriptorProtos;
+import java.time.LocalDateTime;
+import org.apache.spark.sql.types.DataTypes;
+import org.junit.Test;
+
+public class TimestampNTZTypeConverterTest {
+  private final TimestampNTZTypeConverter timestampNTZTypeConverter =
+      new TimestampNTZTypeConverter();
+
+  @Test
+  public void testToSparkType() {
+    assertThat(timestampNTZTypeConverter.toSparkType(LegacySQLTypeName.DATETIME))
+        .isEqualTo(DataTypes.TimestampNTZType);
+  }
+
+  @Test
+  public void testToSparkTypeThrowsException() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> {
+          timestampNTZTypeConverter.toSparkType(LegacySQLTypeName.FLOAT);
+        });
+  }
+
+  @Test
+  public void testToBigQueryType() {
+    assertThat(timestampNTZTypeConverter.toBigQueryType(DataTypes.TimestampNTZType))
+        .isEqualTo(LegacySQLTypeName.DATETIME);
+  }
+
+  @Test
+  public void testToBigQueryTypeThrowsException() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> {
+          timestampNTZTypeConverter.toBigQueryType(DataTypes.TimestampType);
+        });
+  }
+
+  @Test
+  public void testToProtoFieldType() {
+    assertThat(timestampNTZTypeConverter.toProtoFieldType(DataTypes.TimestampNTZType))
+        .isEqualTo(DescriptorProtos.FieldDescriptorProto.Type.TYPE_INT64);
+  }
+
+  @Test
+  public void testToProtoFieldTypeThrowsException() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> {
+          timestampNTZTypeConverter.toProtoFieldType(DataTypes.TimestampType);
+        });
+  }
+
+  @Test
+  public void testSupportsBigQueryType() {
+    assertThat(timestampNTZTypeConverter.supportsBigQueryType(LegacySQLTypeName.DATETIME))
+        .isEqualTo(true);
+    assertThat(timestampNTZTypeConverter.supportsBigQueryType(LegacySQLTypeName.TIMESTAMP))
+        .isEqualTo(false);
+  }
+
+  @Test
+  public void testSupportsSparkType() {
+    assertThat(timestampNTZTypeConverter.supportsSparkType(DataTypes.TimestampNTZType))
+        .isEqualTo(true);
+    assertThat(timestampNTZTypeConverter.supportsSparkType(DataTypes.TimestampType))
+        .isEqualTo(false);
+  }
+
+  @Test
+  public void testSparkToProtoValue() {
+    LocalDateTime javaLocalTime = LocalDateTime.of(2023, 9, 18, 14, 30, 15, 234 * 1_000_000);
+    long protoDateTime = timestampNTZTypeConverter.sparkToProtoValue(javaLocalTime);
+    org.threeten.bp.LocalDateTime threeTenLocalTime =
+        org.threeten.bp.LocalDateTime.of(
+            javaLocalTime.getYear(),
+            javaLocalTime.getMonthValue(),
+            javaLocalTime.getDayOfMonth(),
+            javaLocalTime.getHour(),
+            javaLocalTime.getMinute(),
+            javaLocalTime.getSecond(),
+            javaLocalTime.getNano());
+    assertThat(threeTenLocalTime)
```

Comment:
Let's test for the expected long value instead, without relying on CivilTimeEncoder behaviour.

Reply:
We don't know what the expected long value will be as we are using `CivilTimeEncoder.encodePacked64DatetimeMicros` to encode the LocalDateTime. This is the encoder and decoder used by bigquery

Reply:
Got it.